storage/s3.go

```diff
@@ -658,9 +663,27 @@ func newSession(opts S3Options) (*session.Session, error) {
 		sess.Config.Region = aws.String(endpoints.UsEast1RegionID)
 	}
 
+	s.sessions[opts] = sess
+
 	return sess, nil
 }
 
+// NumOfSessions returns number of sessions currently active.
+func NumOfSessions() int {
```

Comment:
If `NumOfSessions()` and `ClearSessions()` are only intended for TestNumOfSessions test, I think it is better to remove them :) Without them, the code will be much more readable and clean.

Reply:
thanks. they are also used in `TestNewSessionWithRegionSetViaEnv`, when all tests are run together one session created by default region blocks another test from creating a new session with empty region (and set via environment variable).
so, shall i remove them?

Reply:
I am not sure. You may keep these tests if you wish but since these functions are only needed for test suites, I think they should not be part of `storage/s3.go`, maybe they can be added to `storage/s3_test.go`. Also, clearing the global session from test suite is not elegant. If we decide to run these tests parallel in the future, tests will not work since they both share the same global session variable (one may remove session while other trying to access). Unfortunately, I don't have any good solution for now. We may either remove these tests or find a better way to keep separate (nonglobal) session for each test.

Reply:
I am not sure. You may keep these tests if you wish but since these functions are only needed for test suites, I think they should not be part of `storage/s3.go`, maybe they can be added to `storage/s3_test.go`. Also, clearing the global session from test suite is not elegant. If we decide to run these tests parallel in the future, tests will not work since they both share the same global session variable (one may remove session while other trying to access). Unfortunately, I don't have any good solution for now. We may either remove these tests or find a better way to keep separate (nonglobal) session for each test.